src/shortestpaths/floyd-warshall.jl

```diff
@@ -66,6 +60,89 @@ function floyd_warshall_shortest_paths(
     return fws
 end
 
+#Helper function used due to performance bug in @threads. 
+function _loopbody!(
+    pivot::U, 
+    nvg::U,
+    dists::Matrix{T}, 
+    parents::Matrix{U}
+    ) where T<:Real where U<:Integer
+    # Relax dists[u, v] = min(dists[u, v], dists[u, pivot]+dists[pivot, v]) for all u, v
+    @inbounds @threads for v in one(U):nvg
+        d = dists[pivot, v]
+        if d != typemax(T) && v != pivot
+            p = parents[pivot, v]
+            @inbounds for u in one(U):nvg
+                ans = (dists[u, pivot] == typemax(T) || u == pivot ? typemax(T) : dists[u, pivot] + d) 
+                if dists[u, v] > ans
+                    dists[u, v] = ans
+                    parents[u, v] = p
+                end
```

Comment:
How is it there are no race conditions on `dists` here? see specifically the case where `u` == `pivot`.

Reply:
if `u == pivot` or `v == pivot` then that block is not executed.

I made that modification because floyd warshall guarentees correct output (shortest paths) only if
there are no negative cycles. i.e. dists [pivot, pivot] > 0
Hence, that line would not be executed anyway.

Reply:
this prevents `v == pivot`:
` if d != typemax(T) && v != pivot`
this prevents `u == pivot` 
`ans = (dists[u, pivot] == typemax(T) || u == pivot ? typemax(T) : dists[u, pivot] + d)`